DataDefinitions/Engineer.cs

```diff
@@ -93,12 +129,63 @@ public static void AddOrUpdate(Engineer engineer)
             int index = ENGINEERS.FindIndex(candidate => candidate.id == engineer.id);
             if (index != -1)
             {
-                ENGINEERS[index] = engineer;
+                ENGINEERS[index].name = engineer.name;
+                ENGINEERS[index].id = engineer.id;
+                ENGINEERS[index].stage = engineer.stage;
+                ENGINEERS[index].rankprogress = engineer.rankprogress;
+                ENGINEERS[index].rank = engineer.rank;
             }
             else
             {
                 ENGINEERS.Add(engineer);
             }
         }
+
+        public class EngineerSpecialty : ResourceBasedLocalizedEDName<EngineerSpecialty>
```

Comment:
Let's give this class its own .cs file. Makes it easier to find.

Reply:
Agreed.